pkg/controller/deployment/util/deployment_util.go

```diff
@@ -415,14 +415,25 @@ func SetReplicasAnnotations(rs *extensions.ReplicaSet, desiredReplicas, maxRepli
 
 // MaxUnavailable returns the maximum unavailable pods a rolling deployment can take.
 func MaxUnavailable(deployment extensions.Deployment) int32 {
-	if !IsRollingUpdate(&deployment) {
+	if !IsRollingUpdate(&deployment) || *(deployment.Spec.Replicas) == 0 {
```

Comment:
What if deployment's replicas is greater than 0 but smaller than maxUnavailable? Should we return `*(deployment.Spec.Replicas)` if it's smaller?